### ISSUE TITLE
Questions about data sampling in the perturbation tutorial

### ISSUE THREAD

**Issue by YimengQiao** [bowang-lab/scGPT#145]
Hi, thanks for your geat work! In the released perturbation tutorial, I saw you set the max_seq_len=1536 during training stage. I think this would ignore much information. I would like to ask if the perturbation results shown in the paper are also based on models trained with sampling? If so, how did you determine the value of 1536? Thanks a lot!

**Comment by subercui** [comment]
Hi, thank you for the great question. Yes, we reported the results based on this setting with max_seq_len=1536. Note that the model will randomly sample 1536 genes out of the whole input in each iteration, so during training across iterations, it will still see all the input genes and expression. The number 1536 is selected simply based on a balance between training time, memory consumption, and performance. Theoretically, as you said, including more genes during each iteration would potentially increase the performance further. While we did try training with max len equaling all pre-selected ~5500 hvgs on the experiment datasets, interestingly, we found that training on max length of 1536 already provides comparable results while being much faster, so we used this setting.
